### PR TITLE
chore: run tests in isolation

### PR DESCRIPTION
**Pull request by BRKalow**
Builds off of #331

Refactors integration tests to run against isolated fixtures. This is accomplished by copying the fixtures defined in `__tests__/fixtures` into tmp directories and doing a clean install of a locally packed `next-mdx-remote`, along with `next`, `react`, and `react-dom`. This ensures each test case is independently executed and by relying on `npm pack` we are testing against the eventually published artifact.

We now expose a `createDescribe` test utility that handles creating an isolated directory from one of our fixtures:
```ts
import { createDescribe } from '../.jest/utils'

createDescribe('hydration', { fixture: 'basic' }, ({ dir, browser }) => {
	test('something', () => { ... })
})
```

h/t to @timneutkens for pointing me at Next.js's testing approach, which this is inspired by 🙏

### CODE REVIEW

**Review by timneutkens — APPROVED** [review]

**Comment by timneutkens** [comment]
Looks good to me!

File: __tests__/integration.test.tsx [hunk 1]
```diff
@@ -1,201 +1,148 @@
...
-      resolve(undefined)
+    afterAll(async () => {
+      // close the browser and dev server
+      await stopDevServer(childProcess)
     })
-  })
 
-  childProcess.kill('SIGINT')
+    test.each(['/app-dir-mdx/mdxremote', '/app-dir-mdx/compile-mdx'])(
+      'development - MDXRemote - %s',
```
Comment (BRKalow) [inline __tests__/integration.test.tsx:131]:
```suggestion
      'development - RSC - %s',
```

File: __tests__/fixtures/rsc/package.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+  "private": true,
+  "dependencies": {
+    "next": "*",
```
Comment (thiskevinwang) [inline __tests__/fixtures/rsc/package.json:4]: pinning next to `13` might be a good idea, given the [server components](https://nextjs.org/docs/advanced-features/react-18/server-components)
(or at least a comment somewhere stating the expectation that Next >=13 gets used)

Comment (BRKalow) [reply __tests__/fixtures/rsc/package.json]: Updated 👍

File: __tests__/fixtures/rsc/mdx/test.mdx [hunk 3]
```diff
@@ -0,0 +1,32 @@
+---
+title: 'foo'
+name: jeff
```
Comment (thiskevinwang) [inline __tests__/fixtures/rsc/mdx/test.mdx:3]: 😂

File: .jest/utils.tsx [hunk 4]
```diff
@@ -1,28 +1,194 @@
...
+    })
+
+    fn({
+      dir() {
+        return tmpDir
+      },
+      browser() {
+        return browser
+      },
+    })
```
Comment (thiskevinwang) [inline .jest/utils.tsx:106]: neat!

**Review by thiskevinwang — APPROVED** [review]
LGTM!
👏 Very cool new testing pattern here.

**Pull request merged**